src/t_stream.c

```diff
@@ -2188,14 +2188,6 @@ void xreadCommand(client *c) {
         int moreargs = c->argc-i-1;
         char *o = c->argv[i]->ptr;
         if (!strcasecmp(o,"BLOCK") && moreargs) {
-            if (c->flags & CLIENT_SCRIPT) {
-                /*
-                 * Although the CLIENT_DENY_BLOCKING flag should protect from blocking the client
-                 * on Lua/MULTI/RM_Call we want special treatment for Lua to keep backward compatibility.
-                 * There is no sense to use BLOCK option within Lua. */
```

Comment:
so we drop the concern about backwards compatibility? user will get a different error message...

Reply:
even if we do merge this change (not sure what's the justification), i don't think we should backport it to 7.2

Reply:
I see #11568 is considered as improvement rather than breaking change in release note.

Reply:
ok, but i don't think we should backport this to 7.2 (it's not a bug fix, and we don't need to introduce behavior changes in patch level releases)

Reply:
I'm ok with it, let's release in 8.0